src/options/options.js

```diff
@@ -1,3 +1,188 @@
-document.querySelectorAll("[data-locale]").forEach((elem) => {
-    elem.innerText = browser.i18n.getMessage(elem.dataset.locale);
+document.querySelectorAll('[data-locale]').forEach((elem) => {
+  const i18nElement = elem;
+  i18nElement.innerText = browser.i18n.getMessage(i18nElement.dataset.locale);
+});
+
+const sidebarMenuTabs = document.querySelectorAll('input[name="tab-group-1"]');
+const optionsTitle = document.getElementById('options-title');
+const shortcutClearBtnArray = document.querySelectorAll('.clear-button');
+const shortcutResetBtnArray = document.querySelectorAll('.reset-button');
+const inputsShortcutsArr = document.querySelectorAll('.shortcuts-input');
+/*
+Default settings. Initialize storage to these values.
+*/
+const defaultShortCutsArr = [
+  {
+    id: '_execute_browser_action',
+    shortcut: 'Ctrl+Alt+L',
+  },
+];
+
+/*
+Generic error logger.
+*/
+function onError(e) {
+  console.error(e);
+}
+
+function setShortCutsOnLoadPage(storedSettingsArray) {
+  inputsShortcutsArr.forEach((item) => {
+    const shortcutInputElement = item;
+    if (storedSettingsArray) {
+      const getShortcutElement = storedSettingsArray.find((el) => el.id === shortcutInputElement.id);
+      shortcutInputElement.value = getShortcutElement.shortcut;
+      browser.commands.update({
+        name: shortcutInputElement.id,
+        shortcut: getShortcutElement.shortcut,
+      });
+    } else {
+      const getDefaultShortCuts = defaultShortCutsArr.find((el) => el.id === shortcutInputElement.id);
+      browser.storage.local.set({ storedShortCutsArr: defaultShortCutsArr });
+      shortcutInputElement.value = getDefaultShortCuts.shortcut;
+      browser.commands.update({
+        name: shortcutInputElement.id,
+        shortcut: getDefaultShortCuts.shortcut,
+      });
+    }
+  });
+}
+
+/*
+On startup, check whether we have stored settings.
+If we don't, then store the default settings.
+*/
+async function checkStoredSettings(storedSettings) {
+  if (!storedSettings.storedShortCutsArr) {
+    browser.storage.local.set({ storedShortCutsArr: storedSettings.storedShortCutsArr });
+  }
+  await setShortCutsOnLoadPage(storedSettings.storedShortCutsArr);
+}
+
+browser.storage.local.get().then(checkStoredSettings, onError);
+
+sidebarMenuTabs.forEach((item) => {
+  item.addEventListener('click', (e) => {
+    if (e.target.checked) {
+      optionsTitle.innerText = e.target.value;
+    }
+  });
+});
+
+// Clear button handler
+shortcutClearBtnArray.forEach((item) => {
+  const shortcutClearBtnElement = item;
```

Comment:
I think this is redundant

Reply:
It is used at least two times. But...

`shortcutClearBtnElement.previousElementSibling` and `shortcutClearBtnElement.parentNode.nextElementSibling` looks unrobust. What will happen if we will change HTML layout?

It would be more reasonable to use unique `id` for target DOM elements and use `document.getElementById()` to gett access to required element.